### ISSUE TITLE
Specifying usage examples in the spec

### ISSUE THREAD

**Issue by jose-fully-ported** [jdx/usage#3]
The dokku project uses some [faux reflection](https://github.com/dokku/dokku-postgres/blob/master/help-functions) and [fancy footwork](https://github.com/dokku/dokku-postgres/blob/master/bin/generate) to generate cli help output as well as markdown docs [from files](https://github.com/dokku/dokku-postgres/blob/master/subcommands/backup). One thing that is nice is that we can show examples to users for usage.

I'm wondering if that would be possible to somehow spec out. Examples have a `cli command` as well as a `description` of the command, and are shown in the section for that particular subcommand.

**Comment by jdx** [comment]
Those are neat examples, and I think it's related to the problem I kind of need usage for—which might be helpful for you.

I want the ability to "mount" clis into mise-en-place, specifically for tasks so you can do `mise run sql --help` and see output for some defined "task"—which is itself a CLI—but also generate MD docs for the project README and shell completions. Seems like you have a similar problem.

What would be useful, of course, is if we had a library of common CLIs like `psql`. I've actually found at least a couple of good places I think we can use to seed the data—AI might actually help a lot here too.

* https://github.com/withfig/autocomplete/tree/master/src
* https://github.com/sigoden/argc-completions

But I'm not totally sure what you're asking here, are you asking for example usage specs that use some common CLIs so people reading the docs know what to expect? Or are you asking about examples of using a CLI to put into documentation like this:

```
USAGE mycli

DESCRIPITON
....

EXAMPLES
    $ mycli foo
    ...
```

**Comment by jdx** [comment]
there is support for examples: https://github.com/jdx/usage/blob/3f0f16c9b4fc2ff346a97644e97878916c1fa630/src/parse/cmd.rs#L32

though I'm not sure if I have documented it or not

**Comment by josegonzalez** [comment]
Yeah if you could provide an example of usage/output, I can figure out if this would be suitable for my usage or propose spec changes.

**Comment by jdx** [comment]
Commands support `example` nodes, and they render in `--help`, Markdown docs and man pages:

```kdl
cmd "config" help="Manage the CLI config" {
  example "mycli config" header="Show all the config values"
}
```

```
Examples:
  Show all the config values:
    $ mycli config
```

See the `example` section of https://usage.jdx.dev/spec/reference/cmd

_This comment was generated by Claude Code._
